packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1285,6 +1285,7 @@ function handleError(root, thrownValue) {
         // boundary.
         workInProgressRootExitStatus = RootFatalErrored;
         workInProgressRootFatalError = thrownValue;
+        workInProgress = null;
```

Comment:
What is happening here? Why does this work?

Reply:
> This adds a new test case and fixes a bug where React would keep retrying the root because the workInProgress pointer was not advanced to the next fiber. (Which in this case is null, since it's the root.)

Reply:
In other words, it falls into an infinite loop because `workInProgress` is stuck on the the host root fiber forever. Usually `workInProgress` gets advanced during either `beingWork` or `completeUnitOfWork` but since it keeps erroring that never happens.

Reply:
Ah. Subtle. A common would be nice but not required.

Reply:
Ok I'll add one! I think there's a potential follow up to also call `unwindWork` here to pop the contexts. I originally didn't bother because it's not impossible even that call will throw, but if layout errors are common enough it might be worth doing more to prevent corrupted internal state.